### PR TITLE
[RDY]Fixed ":input" screen refresh bug

### PR DESCRIPTION
**Pull request by motiejusvin**
Fixed bug: #14109

### CODE REVIEW

File: src/nvim/ex_cmds.c [hunk 1]
```diff
@@ -2861,12 +2861,14 @@ void ex_append(exarg_T *eap)
...
     for (p = theline; indent > vcol; ++p) {
-      if (*p == ' ')
+      if (*p == ' ') {
         ++vcol;
-      else if (*p == TAB)
+      } else if (*p == TAB) {
         vcol += 8 - vcol % 8;
-      else
+      } else {
+	update_screen(0);
```
Comment (bfredl) [inline src/nvim/ex_cmds.c:2869]: we try to avoid direct calls to `update_screen` if possible. would `redraw_later(curwin, VALID)` work?

Comment (motiejusvin) [reply src/nvim/ex_cmds.c:2869]: No. Screen does not redraw,if replaced `update_screen` to `redraw_later(curwin, VALID)`

Comment (teto) [reply src/nvim/ex_cmds.c:2869]: buffer content changed, shouldn't it be NOT_VALID instead ?

Comment (bfredl) [reply src/nvim/ex_cmds.c:2869]: That redraws everything which is wasteful. I would try `redrawWinline(wp, line)` first.

Comment (motiejusvin) [reply src/nvim/ex_cmds.c:2869]: Fixed. Deleted `update_screen()` and inserted `redrawWinline(curwin,line)` to other place.

Comment (motiejusvin) [reply src/nvim/ex_cmds.c]: What about a for loop,that goes trough every line. Would that be okay in terms of resources?

Comment (motiejusvin) [reply src/nvim/ex_cmds.c]: Fixed. I inserted `redrawWinline(curwin,lnum)` a bit too far. Now its fixed using `redrawWinline(curwin,lnum)`

**Comment by bfredl** [comment]
seems reasonable now :) maybe a test?

**Comment by motiejusvin** [comment]
> seems reasonable now :) maybe a test?

I'm new to this open-source project. So by test it means `make test` ?

**Comment by bfredl** [comment]
yes, `functionaltest` to be more specific. look for tests in `test/functional/ui/*_spec.lua` for examples of screen tests.

**Comment by motiejusvin** [comment]
> yes, `functionaltest` to be more specific. look for tests in `test/functional/ui/*_spec.lua` for examples of screen tests.

Ran some tests. Functionals Without fix had 9 fails,and with had 4 fails.

**Comment by teto** [comment]
when you change code, the linter checks for the neighboring lines of your patch (right before and right after): it allows to refactor the codebase incrementally rather than just refactor everything at once.
see https://github.com/neovim/neovim/pull/14175/checks?check_run_id=2176786800#step:11:225 for the errors. 

Also the errors are diffed against a list of errors on master so it's a good idea to rebase usually.

**Comment by motiejusvin** [comment]
I guess everything's good now after updating the fork

**Comment by teto** [comment]
when you address a linter issue, this changes the line and so the linter looks into the next neighbors, so you can end up changing many lines as it propagates.

**Comment by motiejusvin** [comment]
So, if I understand correctly if I try to fix things, I will just make things worse. And, if I'm done with everything, I just change the title to [RDY]?

**Comment by motiejusvin** [comment]
And thank you for all the time. I'll try to ask less in the future.

**Pull request closed without merging**